deployment/vm/vm.go

```diff
@@ -245,16 +245,24 @@ func (vm *vm) RunScript(script string, options map[string]interface{}) error {
 }
 
 func (vm *vm) Delete() error {
-	deleteErr := vm.cloud.DeleteVM(vm.cid)
-	if deleteErr != nil {
-		// allow VMNotFoundError for idempotency
-		cloudErr, ok := deleteErr.(bicloud.Error)
-		if !ok || cloudErr.Type() != bicloud.VMNotFoundError {
-			return bosherr.WrapError(deleteErr, "Deleting vm in the cloud")
+	_, found, err := vm.vmRepo.FindCurrentIP()
+	if err != nil {
+		return bosherr.WrapError(err, "Finding currently IP address of deployed vm")
+	}
+
+	var deleteErr error
+	if !found {
```

Comment:
Can you add a comment explaining why we don't delete the VM when the `CurrentDynamicIP` is found?

Reply:
@jfmyers9 If SL CPI get vm IP, it can reuse this VM by OS reload. So this VM should not to be truly deleted.

Reply:
@edwardstudy thanks for the explanation. Can you add that as a comment in the code above this if statement? Maybe with slightly more details. Thank you!